src/e3/anod/spec.py

```diff
@@ -387,6 +394,101 @@ def bind_to_sandbox(self, sandbox: SandBox) -> None:
+        ignored_libs: list[str] | None = None,
+        ldd_output: str | None = None,
+    ) -> None:
+        """Sanity check the shared library closure.
+
+        Make sure that the libraries only depend on authorized system shared
+        libraries.
+
+        Raise an error when a problem is detected.
+
+        It is possible to use the *ldd_output* parameter to mimic the ``ldd``
+        call. In that case, the output should look like::
+
+            /a/b/c.so:
+                    d.so => /e/f/d.so
+                    g.so => /a/b/g.so
+            /a/b/h.so:
+                    g.so => /a/b/g.so
+
+        :param prefix: The path where to find the library to be checked.
+        :param ignored_libs: The list of additional system libraries authorized
+            in the closure.
+        :param ldd_output: An ``ldd`` command output.
+
+        :raise: AnodError if some of the shared libraries in *prefix* (or in
+            *ldd_output*) is not in the same directory as the analysed element.
+        """
+        ignored: list[str] = ignored_libs or []
+        errors: dict[str, list[str]] = {}
+        lib_file: str = ""
+
+        if ldd_output is None:
+            # If ldd is not in the path, return and issue a warning.
+            if not which("ldd"):
+                self.log.warning("DLL closure check skipped: ldd could not be found.")
+                return
+
+            if prefix is None:
+                prefix = self.build_space.install_dir
+
+            lib_files = find(
+                root=prefix, pattern=f"*{OS_INFO[e3.env.Env().build.os.name]['dllext']}"
+            )
+            ldd_output = e3.os.process.Run(["ldd"] + lib_files).out or ""
+        else:
+            # An ldd output has been provided.
+            lib_files = re.findall(r"^([^\t].*):$", ldd_output, flags=re.M)
+
+        if self.sandbox and hasattr(self.sandbox, "root_dir"):
+            root_dir = self.sandbox.root_dir
+        else:
+            root_dir = str(Path(self.build_space.install_dir).parent)
+
+        # Retrieve list of shared lib dependencies for sanity checking
+
+        self.log.info("Running DLL closure check")
+        # Build up a list of linked libraries. Dict key is the library
+        # name/path, value is the list of linked libraries.
+        for line in ldd_output.splitlines():
+            if line[:-1] in lib_files:
+                # Line is like ``/mypath/mydll.so:``, it's the result of
+                # ldd for ``mydll.so``.
+                lib_file = line[:-1]
+            elif lib_file and " => " in line:
+                # Line looks like:
+                # ``        otherdll.so => /other/otherdll.so (0xabcd)``
+                name, path = line.strip().split(" => ", 1)
+                path = re.sub(" (.*)", "", path)
+                # Check if we should ignore it. On Windows, the paths are
+                # case-insensitive.
+                if sys.platform != "win32":
```

Comment:
case sensitive is file system dependent, not OS dependent

I would add a parameter with a good default value

Reply:
Good point :) Used to associate case insensitive file system with windows ... My bad.

Reply:
Added the `case_sensitive` parameter.